mws/resource_mws_ncc_private_endpoint_rule.go

```diff
@@ -70,6 +73,52 @@ func ResourceMwsNccPrivateEndpointRule() common.Resource {
 			}
 			return common.StructToData(rule, s, d)
 		},
+		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			nccId, ruleId, err := p.Unpack(d)
+			if err != nil {
+				return err
+			}
+			acc, err := c.AccountClient()
+			if err != nil {
+				return err
+			}
+
+			// only enabled, domain names & resource names are updatable
+			// they do require update_mask to be set
+			// resource_names are not applicable to Azure, so we exclude them from the update
+			updateMask := "enabled"
+			updatePrivateEndpointRule := settings.UpdatePrivateEndpointRule{
+				Enabled: d.Get("enabled").(bool),
+			}
+
+			if d.HasChange("domain_names") {
+				updateMask = "domain_names"
```

Comment:
it should be `updateMask = updateMask + ",domain_names"`, or just use a string array for names, and then do strings.Join before sending the request